### PR TITLE
[candidate_profile] Add card summarizing candidate consent

### PR DESCRIPTION
**Pull request by driusan**
This adds a card with a summary of the candidate's consent to
the candidate_profile module. A table is shown with rows for
each type of consent showing the status and date of the consent
or withdrawal.

### CODE REVIEW

File: modules/candidate_parameters/php/module.class.inc [hunk 1]
```diff
@@ -33,4 +18,57 @@ class Module extends \Module
...
+
+            $candidate = $options['candidate'];
+            if ($candidate === null) {
+                return [];
+            }
+
+            // The candidate getConsents only returns the types of
+            // consents that the candidate has saved, we want to
+            // summarize all of the types configured in the database.
+            $consents = \Utility::getConsentList();
```
Comment (johnsaigle) [inline modules/candidate_parameters/php/module.class.inc]:
```suggestion
    $consents = array_values(
        array_merge(
            \Utility:getConsentList(), 
            $candidate->getConsents()
        )
    );
```

Writing it this way makes it a bit more clear to me that  the $consents array is being sourced from multiple sources. Up to you though.

Comment (driusan) [reply modules/candidate_parameters/php/module.class.inc]: I don't understand what you mean by multiple sources. The consent types are solely sourced from getConsentList and the data from the candidate.

Comment (johnsaigle) [reply modules/candidate_parameters/php/module.class.inc]: Yeah that's what I mean. Anyway it's mostly a cosmetic thing

**Comment by johnsaigle** [comment]
Screen Shot 2020-03-03 at 12 15 34

The new card appears - data may be missing on Raisinbread or my VM is not well-configured.

On a related note: what's the expected route to get to candidate_profile? I wasn't sure so I entered the URL manually.

File: modules/candidate_parameters/jsx/ConsentWidget.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            <th>Date</th>
+        </tr>
+        </thead>
+        <tbody>
+        {consents}
+        </tbody>
+    </table>);
+}
+
+function consentTerm(consent) {
```
Comment (johnsaigle) [inline modules/candidate_parameters/jsx/ConsentWidget.js]: A function comment would be good

**Comment by driusan** [comment]
There's no frontend route to access it because it's not ready yet, you need to manually enter the URL (`$LORIS/candidate_profile/$candID`) to test a PR.. the screenshot looks like consent is configured for the dataset, but there's no consent data for that candidate in your dataset. You can edit/add some through the candidate_parameters module on your VM.

**Comment by johnsaigle** [comment]
Screen Shot 2020-03-03 at 15 12 53
Looks good

**Pull request merged**